integration/presto/presto-integration-technical-note.md

```diff
@@ -0,0 +1,253 @@
+2. Copy `carbondata` jars to `plugin/carbondata` directory on all nodes.
+
+### Start Presto Server on all nodes
+
+```
+./presto-server-0.187/bin/launcher start
+```
+To run it as a background process.
+
+```
+./presto-server-0.187/bin/launcher run
+```
+To run it in foreground.
+
+### Start Presto CLI
+```
+./presto
+```
+To connect to carbondata catalog use the following command:
+
+```
+./presto --server <coordinator_ip>:8086 --catalog carbondata --schema <schema_name>
+```
+Execute the following command to ensure the workers are connected.
+
+```
+select * from system.runtime.nodes;
+```
+Now you can use the Presto CLI on the coordinator to query data sources in the catalog using the Presto workers.
+
+**Note :** Create Tables and data loads should be done before executing queries as we can not create carbon table from this interface.
+
+## **Presto Performance Tuning**
+
+**Performance Optimizations according to data types and schema:**
+
+- When the data could be stored in Int as well as String. Example: keys for a table then using Int gives a better performance
+
+- Use Double instead of Decimal if required precision is low.
+
+- Columns having low-cardinality should be created as dictionary columns. This will improve query performance to a great extent.
+
+**Performance Optimization by changing Queries:**
+
+- There’s a probability where GROUP BY becomes a little bit faster, by carefully ordering a list of fields within GROUP BY in an order of high cardinality.
+
+- Aggregating a series of LIKE clauses in one single regexp_like clause.
+
+**For example :**
+
+```
+SELECT
+  ...
+FROM
+  access
+WHERE
+  method LIKE '%GET%' OR
+  method LIKE '%POST%' OR
+  method LIKE '%PUT%' OR
+  method LIKE '%DELETE%'
+ ```
+
+ can be optimized by replacing the 4 LIKE clauses with a single regexp_like clause:
+
+ ```
+ SELECT
+  ...
+FROM
+  access
+WHERE
+  regexp_like(method, 'GET|POST|PUT|DELETE')
+ ```
+
+- Specifying large tables first in join clause
+
+
+The default join algorithm of Presto is broadcast join, which partitions the left-hand side table of a join and sends (broadcasts) a copy of the entire right-hand side table to all of the worker nodes that have the partitions. This works when your right table is small enough to fit within one node (usually less than 2GB). If you observe ‘Exceeded max memory xxGB’ error, this usually means the right-hand side table is too large. Presto does not perform automatic join-reordering, so please make sure your large table preceeds small tables in any join clause.
+
+**Note :** If you still see the memory issue, try distributed hash join. This algorithm partitions both the left and right tables using the hash values of the join keys. So the distributed join would work even if the right-hand side table is large, but the performance can be slower because it increases the number of network data transfers. To turn on the distributed join, embed the following session property as an SQL comment:
```

Comment:
can we specify how to use distributed has join

Reply:
done